### ISSUE TITLE
call.uaddr runtime test failing on RHEL

### ISSUE THREAD

**Issue by dubeyabhishek** [bpftrace/bpftrace#3025]
# ./runtime-tests.sh --filter 'call.uaddr'
====================
bpftrace --info:
====================
System
  OS: Linux 5.14.0-423.el9.ppc64le #1 SMP Sat Feb 17 10:31:43 EST 2024
  Arch: ppc64le

Build
  version: v0.19.1
  LLVM: 17.0.6
  unsafe probe: no
  bfd: yes
  libdw (DWARF support): yes

Kernel helpers
  probe_read: yes
  probe_read_str: yes
  probe_read_user: yes
  probe_read_user_str: yes
  probe_read_kernel: yes
  probe_read_kernel_str: yes
  get_current_cgroup_id: yes
  send_signal: yes
  override_return: no
  get_boot_ns: yes
  dpath: yes
  skboutput: no
  get_tai_ns: yes
  get_func_ip: yes

Kernel features
  Instruction limit: 1000000
  Loop support: yes
  btf: yes
  module btf: yes
  map batch: yes
  uprobe refcount (depends on Build:bcc bpf_attach_uprobe refcount): yes

Map types
  hash: yes
  percpu hash: yes
  array: yes
  percpu array: yes
  stack_trace: yes
  perf_event_array: yes
  ringbuf: yes

Probe types
  kprobe: yes
  tracepoint: yes
  perf_event: yes
  kfunc: no
  kprobe_multi: no
  raw_tp_special: yes
  iter: yes

[==========] Running 1 tests from 1 test cases.

[----------] 1 tests from call
[ RUN      ] call.uaddr
[  FAILED  ] call.uaddr
	Command: ../src//bpftrace -e 'uprobe:testprogs/uprobe_test:function1 { printf("0x%lx -- 0x%lx\n", *uaddr("GLOBAL_A"), *uaddr("GLOBAL_C")); exit(); }' -c ./testprogs/uprobe_test
	Expected: 0x55555555 -- 0x33333333
	Found: Attaching 1 probe...\n__BPFTRACE_NOTIFY_PROBES_ATTACHED\n0x0 -- 0x0\n\n\n
[----------] 1 tests from call

[==========] 1 tests from 1 test cases ran. (1.5623672008514404 total)
[  PASSED  ] 0 tests.
[  FAILED  ] 1 tests, listed below:
[  FAILED  ] call.uaddr

@viktormalik @danobi Why GLOBAL_C/A read as zero? Is that some compiler limitation?

**Comment by viktormalik** [comment]
> @viktormalik @danobi Why GLOBAL_C/A read as zero? Is that some compiler limitation? I have already disabled ASLR in /etc/sysctl.conf kernel.randomize_va_space = 0

This shouldn't be a compiler issue.

Does this happen on ppc64le RHEL only? Could you try to reproduce the issue on x86 or a different distro?

Also could you give the output of `file testprogs/uprobe_test`?

Thanks!

**Comment by dubeyabhishek** [comment]
Yes, its only happening on ppc64le with RHEL. Other tests that are failing on RHEL:
[  FAILED  ] builtin.func_uprobe_elf_symtable
[  FAILED  ] call.ustack_elf_symtable

Same are passing on Fedora 39 on ppc64le.

**Comment by viktormalik** [comment]
What about x86 RHEL (preferably in the same version)? Would you have an option to test that?

**Comment by dubeyabhishek** [comment]
Unfortunately, I don't have x86 RHEL to test on. I tested bpftrace on centOS-9-stream(upstream of RHEL 9.x) and I have completely new set of failures coming in, listed below:
[  FAILED  ] 14 tests, listed below:
[  FAILED  ] probe.kprobe_offset_fail_size
[  FAILED  ] uprobe.uprobes - list probes by pid
[  FAILED  ] uprobe.uprobes - list probes by pid in separate mount namespace
[  FAILED  ] usdt.usdt - list probes by file with wildcarded probe type
[  FAILED  ] usdt.usdt probes - all probes by wildcard and file with child
[  FAILED  ] usdt.usdt probes - attach to fully specified probe of child
[  FAILED  ] usdt.usdt probes - attach to probe by wildcard and file with child
[  FAILED  ] usdt.usdt probes - attach to probe on multiple files by wildcard
[  FAILED  ] usdt.usdt probes - attach to probe with probe builtin and args by file with child
[  FAILED  ] usdt.usdt probes - attach to probes by wildcard file with child
[  FAILED  ] usdt.usdt probes - list probes by pid in separate mountns
[  FAILED  ] usdt.usdt sized arguments

**Comment by viktormalik** [comment]
Unfortunately, I'm not able to reproduce with the latest bpftrace:
```
# tests/runtime-tests.sh --filter 'call.uaddr'
====================
bpftrace --info:
====================
System
  OS: Linux 5.14.0-427.el9.ppc64le #1 SMP Fri Feb 23 01:42:26 EST 2024
  Arch: ppc64le

Build
  version: v0.20.0-58-g7c98-dirty
  LLVM: 17.0.6
  unsafe probe: no
  bfd: yes
  libdw (DWARF support): yes

[...]

[==========] Running 1 tests from 1 test cases.

[----------] 1 tests from call
[ RUN      ] call.uaddr
[       OK ] call.uaddr
[----------] 1 tests from call

[==========] 1 tests from 1 test cases ran. (1419 ms total)
[  PASSED  ] 1 tests.
```
Similarly, most of the above mentioned tests pass, the only failures that I can see are:
```
[  FAILED  ] 5 tests, listed below:
[  FAILED  ] usdt.usdt probes - attach to probe with semaphore
[  FAILED  ] usdt.usdt probes - file based semaphore activation
[  FAILED  ] usdt.usdt probes - file based semaphore activation
[  TIMEOUT ] probe.kprobe_offset_fail_size
[  TIMEOUT ] usdt.usdt probes - file based semaphore activation no process, kernel usdt semaphore
```
@dubeyabhishek could you please check if you can reproduce with the latest bpftrace from GitHub? Thanks!

**Comment by dubeyabhishek** [comment]
@viktormalik I got the reported(this issue) failure on v0.19.1-bpftrace
Other failures in the list were:
[  FAILED  ] 5 tests, listed below:
[  FAILED  ] builtin.func_uprobe_elf_symtable
[  FAILED  ] call.uaddr
[  FAILED  ] call.ustack_elf_symtable
[  FAILED  ] probe.kprobe_offset_module_error
[  FAILED  ] usdt.usdt probes - attach to fully specified probe all pids

Possibly, the latest bpftrace may have different failure set on RHEL. I will check and report.
